src/main/java/entities/BotPlayer.java

```diff
@@ -1,93 +1,123 @@
-
-      // As a bot, we generally want to save our wild cards for later.
-      // Number (6), Reverse (5), Skip (4), Wild(3), Draw Two (2), Draw Four (1)
-      if (cardPriority > bestMovePriority) {
-        bestMovePriority = cardPriority;
-        bestMove = i;
-      }
-    }
-
-    return bestMove;
-  }
-
-  public int getRandomMove(Card lastPlayed) {
-    ArrayList<Card> possibleMoves = this.getPossibleMoves(lastPlayed);
-    int numberOfPossibleMoves = possibleMoves.size();
-
-    if (numberOfPossibleMoves == 0) {
-      return -1;
-    }
-
-    Random rand = new Random();
-    int n = rand.nextInt(possibleMoves.size());
-
-    return n;
-  }
-  
-  /**
-   * Is the next player about to Uno?
-   * @param nextPlayerCards
-   */
-  private int upsetNextPlayerWeight(ArrayList<Card> nextPlayerCards) {
-    if (nextPlayerCards.size() <= 2) {
-      return 10;
-    }
-
-    return 0;
-  }
+	private final static String playerType = "bot";
+	private String difficulty;
+
+	public BotPlayer(String name, String difficulty) {
+		super(name);
+		this.difficulty = difficulty;
+	}
+
+	public String getPlayerType() {
+		return playerType;
+	}
+
+	public int getBestMove(Card lastPlayed, List<Card> nextPlayerHand) {
+		List<Card> possibleMoves = this.getPossibleMoves(lastPlayed);
+
+		int bestMove = -1;
+		int bestMovePriority = 0;
+
+		for (int i = 0; i < possibleMoves.size(); i++) {
+			Card card = possibleMoves.get(i);
+
+			int cardPriority = card.getBotPriority();
+
+			// Make sure the next player doesn't win
+			if (Arrays.asList("Plus4", "Plus2", "Skip", "Reverse").contains(card.getCardType())) {
+				cardPriority += upsetNextPlayerWeight(nextPlayerHand);
+			}
+
+			if (cardPriority > bestMovePriority) {
+				bestMovePriority = cardPriority;
+				bestMove = i;
+			}
+		}
+
+		return bestMove;
+	}
+
+	public int getMovesFromWeights(Card lastPlayed) {
+		List<Card> possibleMoves = this.getPossibleMoves(lastPlayed);
+		int bestMove = -1;
+		int bestMovePriority = 0;
+
+		for (int i = 0; i < possibleMoves.size(); i++) {
+			Card card = possibleMoves.get(i);
+
+			// pass is next player uno?
+			int cardPriority = card.getBotPriority();
+
+			// As a bot, we generally want to save our wild cards for later.
+			// Number (6), Reverse (5), Skip (4), Wild(3), Draw Two (2), Draw Four (1)
+			if (cardPriority > bestMovePriority) {
+				bestMovePriority = cardPriority;
+				bestMove = i;
+			}
+		}
+
+		return bestMove;
+	}
+
+	public int getRandomMove(Card lastPlayed) {
+		List<Card> possibleMoves = this.getPossibleMoves(lastPlayed);
+		int numberOfPossibleMoves = possibleMoves.size();
+
+		if (numberOfPossibleMoves == 0) {
+			return -1;
+		}
+
+		Random rand = new Random();
+		int n = rand.nextInt(possibleMoves.size());
+
+		return n;
+	}
+
+	/**
+	 * Is the next player about to Uno?
+	 * 
+	 * @param nextPlayerCards
+	 */
+	private int upsetNextPlayerWeight(List<Card> nextPlayerCards) {
+		if (nextPlayerCards.size() <= 2) {
```

Comment:
I've got a cool way of rewriting this line. Using the ternary operator, you can write upsetNextPlayer() as "return nextPlayerCards.size() <= 2 ? 10 : 0"